userspace/libsinsp/container_engine/rkt.cpp

```diff
@@ -215,7 +215,10 @@ bool rkt::rkt::parse_rkt(sinsp_container_info &container, const string &podid, c
 		container.m_image = jroot["name"].asString();
 		for(const auto& label_entry : jroot["labels"])
 		{
-			container.m_labels.emplace(label_entry["name"].asString(), label_entry["value"].asString());
+			string val = label_entry["value"].asString();
```

Comment:
not sure what assstring does, but we're doing it twice here, so may be non-ideal if it's creating a new object/copyin ght estring.

Also, do we run into issues with a null terminator here? I don't know enough about these functions to know if it should be < instead of <=

Reply:
`AsString` is simply converting a `Json::Value` object to `std::string` and no we're not doing it twice, it is doing it once on `value` Json object and creating a string. I am storing it away in a local copy to precisely prevent it from having to do it twice. 

Also, here the `size` or `length` functions of std::string always return only the actual chars represented, even if the underlying storage has `\0` in there. So it is correct to do `<=` --- but bigger picture it really doesn't matter all that much except allowing labels of a particular size and not allowing a label of one size higher. So it just draws a line _somewhere_  - in the grand scheme of things it doesn't matter if we are off by 1.

Reply:
"simply converting" is often tantamount to making a copy of the actual backing data.